### PR TITLE
Remove DisplayMode

### PR DESCRIPTION
**Pull request by EdAyers**
The only thing that DisplayMode was doing was changing the title of infoview and deciding whether or not to display the 'all goals' collapsible section of the infoview. So I think it makes sense to remove it and to also remove the distinction between "Display Goals" and "Display Messages". 

Also remove config to do with filtering hypotheses.

### CODE REVIEW

**Comment by EdAyers** [comment]
I'll ask on zulip whether this is sensible

File: package.json [hunk 1]
```diff
@@ -98,65 +98,11 @@
...
-					"type": "boolean",
-					"default": true,
-					"description": "Info view: auto open shows goal and messages for the current line (instead of all messages for the whole file)"
-				},
 				"lean.infoViewStyle": {
 					"type": "string",
 					"default": "",
 					"description": "Add an additional CSS snippet to the info view."
 				},
-				"lean.infoViewTacticStateFilters": {
```
Comment (gebner) [inline package.json:111]: These still work.

File: package.json [hunk 2]
```diff
@@ -366,18 +276,6 @@
 				"key": "alt+v",
 				"mac": "alt+v",
 				"when": "editorTextFocus && editorLangId == lean"
-			},
-			{
-				"command": "lean.infoView.displayGoal",
-				"key": "ctrl+shift+enter",
```
Comment (gebner) [inline package.json:372]: This command (and in particular the keycombo) should continue to work.

File: package.json [hunk 3]
```diff
@@ -98,65 +98,11 @@
 					"default": true,
 					"description": "Info view: open info view when Lean extension is activated."
 				},
-				"lean.infoViewAutoOpenShowGoal": {
```
Comment (gebner) [inline package.json:101]: This should remain configurable.

File: package.json (hunk 2 above)
Comment (EdAyers) [reply package.json:372]: Yes it does still work using `lean.displayGoal` instead of `lean.infoview.diaplayGoal`

File: infoview/index.tsx [hunk 4]
```diff
@@ -94,7 +94,7 @@ function Main(props: {}) {
                 const isCursor = locationEq(loc,curLoc.loc);
                 return <Info loc={loc} isPaused={paused} setPaused={setPause(i)} key={i} isPinned={true} isCursor={isCursor} onEdit={onEdit} onPin={unpin(i)}/>}) }
             {!isPinned(curLoc.loc) && <Info loc={curLoc.loc} isPaused={curLoc.paused} setPaused={setPause()} key={pinnedLocs.length} isPinned={false} isCursor={true} onEdit={onEdit} onPin={pin}/>}
-            <details>
+            <details open={!config.infoViewAutoOpenShowGoal}>
```
Comment (gebner) [inline infoview/index.tsx:97]: Is the `<Messages/>` here rendered always, even if collapsed?  Is this a performance issue?

Comment (EdAyers) [reply infoview/index.tsx:97]: Yes it is rendered always. Afaik it's not a perf issue.

**Pull request merged**